drafts/str4d-overwinter-activation/draft1.rst

```diff
@@ -0,0 +1,237 @@
+activation mechanism is necessary so that nodes can unambiguously know at what point the changes from an
+upgrade come into effect (and can enforce consensus rule changes, for example).
+
+
+Specification
+=============
+
+The following constants are defined for every network upgrade:
+
+BRANCH_ID
+  A globally-unique non-zero 32-bit identifier.
+
+  Implementations MAY use a value of zero in branch ID fields to indicate the absence of any upgrade (i.e.
+  that the Sprout consensus rules apply).
+
+ACTIVATION_HEIGHT
+  The non-zero block height at which the network upgrade rules will come into effect, and be enforced as part
+  of the blockchain consensus.
+
+  For removal of ambiguity, the block at height ``ACTIVATION_HEIGHT - 1`` is subject to the pre-upgrade
+  consensus rules, and would be the last common block in the event of a persistent pre-upgrade branch.
+
+  It MUST be greater than the value of ``DEPRECATION_HEIGHT`` in the last software version that will not
+  contain support for the network upgrade. It SHOULD be chosen to be reached approximately three months after
+  the first software version containing support for the network upgrade is released, for the following reason:
+
+  - As of the time of writing (the 1.0.15 release), the release cycle is six weeks long, and nodes undergo
+    auto-senescence 16 weeks after release. Thus, if version ``X`` contains support for a network upgrade,
+    version ``X-1`` will deprecate 10 weeks after the release of version ``X``, which is about 2.3 months. A
+    three-month window provides ample time for users to upgrade their nodes after auto-senescence, and
+    re-integrate into the network prior to activation of the network upgrade.
+
+The relationship between ``BRANCH_ID`` and ``ACTIVATION_HEIGHT`` is many-to-one: it is possible for many
+distinct branches to descend from the same parent block (and thus have the same ``ACTIVATION_HEIGHT``), but a
+specific branch can only have one parent block. Concretely, this means that if the ``ACTIVATION_HEIGHT`` of a
+network upgrade is changed for any reason (e.g. security vulnerabilities or consensus bugs are discovered),
+the ``BRANCH_ID`` MUST also be changed.
+
+Activation mechanism
+--------------------
+
+A blockchain is defined as invalid if, within the set of all network upgrades that have activated in the past
+(or will activate in future) on that chain, an ``ACTIVATION_HEIGHT`` is repeated. Note that this does not
+require ``ACTIVATION_HEIGHT`` to be globally unique, or even locally unique; multiple network upgrades can
+occur in parallel, as long as they are non-overlapping (only one will activate on any given chain).
+
+Concretely, this means that the Zcash blockchain is broken into "epochs" of block height intervals
+``[ACTIVATION_HEIGHT_N, ACTIVATION_HEIGHT_{N+1})`` (ie. including ``ACTIVATION_HEIGHT_N`` and excluding
+``ACTIVATION_HEIGHT_{N+1}``), on which consensus rule sets are defined.
```

Comment:
Almost, but not quite. An epoch is a specific subset of a branch between its activation block, and the activation block of the next epoch in the blockchain. The branch itself continues to exist conceptually beyond the epoch, and may exist concretely in the case of a contentious upgrade.

If you have a suggestion for how to improve this definition, I'd love to hear it. I defined it down here rather than up in the terminology section because it depends on values introduced in this ZIP, rather than more general blockchain concepts.